### PR TITLE
SHIPAPPS-98 Upgrade ship-components-textinput to React 16

### PR DESCRIPTION
**Pull request by adamsas**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,10 +67,9 @@
     "postcss-loader": "^0.7.0",
     "postcss-nested": "^1.0.0",
     "postcss-simple-vars": "^1.1.0",
-    "react": "^15.6.2",
-    "react-addons-css-transition-group": ">=15.0.0",
-    "react-dom": "^15.6.2",
-    "react-test-renderer": "^15.6.2",
+    "react": "^16.6.3",
```
Comment (JaredBoone) [inline package.json]: Maybe we should change these react* version dependencies to ">=16.0.0"

File: tasks/config/webpack.config.js [hunk 2]
```diff
@@ -20,8 +20,7 @@ module.exports = {
     'react': 'React',
     'react-dom': true,
     'react-highlight-click': true,
-    'react-outsideclick': true,
-    'react-addons-css-transition-group': true
+    'react-outsideclick': true
```
Comment (JaredBoone) [inline tasks/config/webpack.config.js]: I don't think we can remove this. See Slideshow.js: 
`const Transition = require('react-addons-css-transition-group');`

File: dist/TextInput.css [hunk 3]
```diff
@@ -85,7 +85,7 @@
 }
 
 .text-input--error {
-  font-size: 12px;
+  font-size: inherit;
```
Comment (JaredBoone) [inline dist/TextInput.css]: @adamsas Please tell me about this change

Comment (adamsas) [reply dist/TextInput.css]: @JaredBoone This is likely from a previous change that was not built into dist. It shouldn't matter as dist is now ignored too.

File: package.json (hunk 1 above)
Comment (isuttell) [reply package.json]: I'm wondering if we need to limit it to <17 as well to protect ourselves

Comment (JaredBoone) [reply package.json]: Agreed

Comment (JaredBoone) [reply package.json]: ">=16.0.0 <17.0.0"

**Review by isuttell — APPROVED** [review]

**Review by JaredBoone — APPROVED** [review]

**Pull request merged**